src/commands/install-packages.yml

```diff
@@ -72,10 +72,11 @@ steps:
                 echo "Found yarn.lock file, assuming lockfile"
                 cp yarn.lock /tmp/node-project-lockfile
               fi
+              cp package.json /tmp/node-project-package.json
         - restore_cache:
             keys:
-              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "package.json"}}-{{ checksum "/tmp/node-project-lockfile" }}
-              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "package.json"}}-
+              - node-deps-{{ arch }}-<<parameters.cache-version>>-<<#parameters.include-branch-in-cache-key>>{{ .Branch }}-<</parameters.include-branch-in-cache-key>>-{{ checksum "/tmp/node-project-package.json"}}-{{ checksum "/tmp/node-project-lockfile" }}
```

Comment:
There is an extra hyphen after `<</parameters.include-branch-in-cache-key>>`, which is a user-visible weirdness that may frustrate those trying to manually restore these caches. Less importantly, there is a missing space after `package.json"`. (Clever though; it wouldn't have occurred to me to include the package.json checksum among the fallback strategies.)

Reply:
Having reflected upon this further, including package.json in the cache key will help projects that frequently upgrade only transitive dependencies at the expense of projects that frequently upgrade their own package version (or otherwise modify package.json) without modifying any dependencies. I am not sure whether it is a net benefit or not.